### PR TITLE
DOCS-3630 ASM terminology page

### PR DESCRIPTION
**Pull request by kayayarai**
### What does this PR do?
Adds a glossary of terms and references for ASM

### Motivation
DOCS-3630

### Additional Notes

This has been signed off on by the PM, and the definitions have gone through SME review, and my edit pass. Light review needed only.

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by sourabhkatti** [comment]
Can we remove the "Web Development Terms" section? https://docs-staging.datadoghq.com/kari/docs-3630-asm-terms/security/application_security/terms/#web-development-terms

Looking at it in this list, I don't think we need to include them since they're fairly generic and not specific to ASM

File: content/en/security/application_security/terms.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+[1]: https://owasp.org/www-community/vulnerabilities/
+[2]: https://owasp.org/www-project-top-ten/
+[3]: https://owasp.org/www-community/attacks/xss/
+[4]: https://owasp.org/www-community/attacks/SQL_Injection
+[5]: https://owasp.org/Top10/A10_2021-Server-Side_Request_Forgery_%28SSRF%29/
+[6]: https://owasp.org/www-project-web-security-testing-guide/v42/4-Web_Application_Security_Testing/07-Input_Validation_Testing/11.1-Testing_for_Local_File_Inclusion
+[7]: https://owasp.org/www-project-top-ten/2017/A1_2017-Injection
+[8]: /agent/guide/how_remote_config_works/
+[9]: https://en.wikipedia.org/wiki/Software_framework
```
Comment (kayayarai) [inline content/en/security/application_security/terms.md:133]:
```suggestion
```

**Comment by buraizu** [comment]
Thanks, created DOCS-4896 to review

File: content/en/security/application_security/terms.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+Remote Configuration
+: A Datadog platform mechanism that enables the Agent configuration to be updated remotely. Used by ASM to update event rules, activate the product, and block attackers.
+: See [How Remote Configuration Works][8].
+
+service
+: A single web application, microservice, API, or function. Usually serves a business function.
+
+signal
+: A detection of an application attack that impacts your services. Signals identify meaningful threats for you to review, and should be triaged with a high priority.
+: See [Signals Explorer][13] in the app
```
Comment (alai97) [inline content/en/security/application_security/terms.md:52]:
```suggestion
: See [Signals Explorer][13] in the app.
```

File: content/en/security/application_security/terms.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+: Comparing the open source libraries loaded by your services to databases of known vulnerabilities. SCA helps you identify vulnerable dependencies, outdated libraries, and licensing issues in the open source libraries that your web services load.
+
+severity
+: An indicator of how quickly an attack attempt should be triaged and addressed. Based on a combination of factors, including the attack's potential impact and risk. Values are Critical, High, Medium, Low, Info.
+
+suspicious request 
+: A distributed trace for which security activity has been flagged by event rules. The underlying trace is shared with APM, allowing deeper and faster investigations. 
+
+user attribution
+: A mechanism that maps suspicious requests to known users your systems.
```
Comment (alai97) [inline content/en/security/application_security/terms.md:64]:
```suggestion
: A mechanism that maps suspicious requests to known users in your systems. 
```

File: content/en/security/application_security/terms.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+Structured Query Language Injection (SQLi, SQL Injection)
+: A type of injection attack in which a SQL query is executed via the input data from the client to the application. SQL commands are injected into data-plane input in order to affect the execution of predefined SQL commands. A successful SQL injection exploit can read sensitive data from the database, modify database data (Insert/Update/Delete), execute administration operations on the database (such as shutdown the DBMS), recover the content of a given file present on the DBMS file system, and in some cases issue commands to the operating system.
+: **Related**: Cassandra Query Language Injection (CQLi), NoSQL Injection (NoSQLi) - Similar to SQLi but for the Cassandra Query Language and NoSQL. 
+: See [SQL Injection on OWASP][4].
+
+Server-Side Request Forgery (SSRF)
+: A vulnerability where a web application fetches a remote resource without validating the user-supplied URL. It allows an attacker to coerce the application to send a crafted request to an unexpected destination, even when protected by a firewall, VPN, or another type of network access control list (ACL).: See [Server-Side Request Forgery on OWASP][5].
```
Comment (alai97) [inline content/en/security/application_security/terms.md:85]:
```suggestion
: A vulnerability where a web application fetches a remote resource without validating the user-supplied URL. It allows an attacker to coerce the application to send a crafted request to an unexpected destination, even when protected by a firewall, VPN, or another type of network access control list (ACL).
: See [Server-Side Request Forgery on OWASP][5].
```

File: content/en/security/application_security/terms.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+: See [Testing for LFI on OWASP][6].
+
+Remote File Inclusion (RFI)
+: A vulnerability similar to Local File Inclusion, but allows an attacker to include a remote file during the processing of the request. The files used in Remote File Inclusion attacks most commonly contain malicious code for PHP, JSP, or similar technologies.
+
+Remote Code Execution (RCE)
+: A vulnerability that allows an attacker to remotely execute code on a machine.
+
+Object-Graph Navigation Language Injection (OGNLi)
+: A vulnerability that allows an attacker to execute their own OGNL expression in a Java application, most commonly leading to remote code execution
```
Comment (alai97) [inline content/en/security/application_security/terms.md:98]:
```suggestion
: A vulnerability that allows an attacker to execute their own OGNL expression in a Java application, most commonly leading to remote code execution.
```

**Review by alai97 — APPROVED** [review]
Looks good, some copy suggestions!

**Pull request merged**